build/nested-cluster/start.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+wait_for_docker ()
+{
+  # Wait for docker.
+  until docker version; do sleep 1 ;done
+}
+
+start_kubelet ()
+{
+  # Start the kubelet.
+  mkdir -p /etc/kubernetes/manifests
+  mkdir -p /etc/srv/kubernetes
+
+  # Change the kubelet to not fail with swap on.
+  cat > /etc/systemd/system/kubelet.service.d/kubeadm-20.conf << EOM
+[Service]
+Environment="KUBELET_EXTRA_ARGS=--fail-swap-on=false"
+EOM
+#  sed -i 's/ExecStart=\/usr\/bin\/kubelet/ExecStart=\/usr\/bin\/kubelet --fail-swap-on=false/' /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+  systemctl enable kubelet
+  systemctl start kubelet
+}
+
+start_worker ()
+{
+  wait_for_docker
+  start_kubelet
+
+  # Load docker images
+  docker load -i /kube-proxy-amd64.tar
+
+  # Start kubeadm.
+  /usr/bin/kubeadm join --token=abcdef.abcdefghijklmnop --discovery-token-unsafe-skip-ca-verification=true --ignore-preflight-errors=all 172.18.0.2:6443 2>&1
+}
+
+start_master ()
+{
+  wait_for_docker
+  mount --make-rshared /etc/kubernetes
+  start_kubelet
+
+  # Load the docker images
+  docker load -i /cloud-controller-manager-amd64.tar
+  docker load -i /kube-apiserver-amd64.tar
+  docker load -i /kube-controller-manager-amd64.tar
+  docker load -i /kube-proxy-amd64.tar
+  docker load -i /kube-scheduler-amd64.tar
+
+  # Run kubeadm init to config a master.
+  /usr/bin/kubeadm init --token=abcdef.abcdefghijklmnop --ignore-preflight-errors=all --kubernetes-version=$(/k8sversion) --pod-network-cidr=192.168.0.0/16 --apiserver-cert-extra-sans $1 2>&1
+
+  # We'll want to read the kube-config from outside the container, so open read
+  # permissions on admin.conf.
+  chmod a+r /etc/kubernetes/admin.conf
+
+  # We need to prevent kube-config from trying to set conntrack values.
+  kubectl --kubeconfig=/etc/kubernetes/admin.conf get ds -n kube-system kube-proxy -o json | jq '.spec.template.spec.containers[0].command |= .+ ["--conntrack-max-per-core=0"]' | kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f -
+  # Apply a pod network.
+  # Calico is an ip-over-ip overlay network. This saves us from many of the
+  # difficulties from configuring an L2 network.
+  kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f http://docs.projectcalico.org/v2.4/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml
```

Comment:
If this is to be used for testing, I don't think hardcoding the network addon is a good idea.

Reply:
It's going to be used moreso for integration style testing, the network addons should be tested in other ways. I don't think there is any intention to support testing them here.

Reply:
Hmm fair enough. Maybe something to PR in a follow-up then? 

I do think we probably want to avoid this being commonly configured though, so we can make it easy for users to know that locally running tests and the CI tests match etc.

Reply:
Although we don't intend to test every network provider ourselves, it's reasonable to use this to test your provider. Calico was chosen because it worked. If you'd like to get Weave working, that would be awesome.

And yes, depending on the external URL bothers me. But for now it sounds better than maintaining overlay configs. Until we have a better way to consume addons, or unless we see failures due to it, we'll do something similar for metrics-server, kubernetes-dashboard, etc.